.github/workflows/backend_unit_tests.yml

```diff
@@ -0,0 +1,33 @@
+name: Backend unit tests
+
+on:
+  push:
+    branches: [main]
+  pull_request: {}
+
+jobs:
+  pytest:
+    permissions: write-all
+    # environment: development
+    runs-on: ubuntu-latest
+    env:
+      COHERE_API_KEY: ${{ secrets.COHERE_API_KEY }}
+      DATABASE_URL: ${{ secrets.DATABASE_URL }}
+      NEXT_PUBLIC_API_HOSTNAME: ${{ secrets.NEXT_PUBLIC_API_HOSTNAME }}
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Setup Docker
+        run: docker compose up test_db -d
+      - name: Run unit tests
+        if: github.actor != 'dependabot[bot]'
+        run: |
+          mkdir -p coverage
+          docker compose build --build-arg INSTALL_COMMUNITY_DEPS=true backend
```

Comment:
Do unit tests still need an service to run? Could we just not run them independently

Reply:
I think you're asking if we could run the tests without needing to build the docker container, right?

That is how I run single tests or single test files locally, so it's certainly achievable. I see that our tooling also builds a container and runs the tests there when we use the `make run-unit-tests` command, so I'm guessing whoever set this workflow up decided to do things here the same way either for consistency or for ease of implementation.